src/ftxui/component/input.cpp

```diff
@@ -101,7 +101,8 @@ class InputBase : public ComponentBase, public InputOption {
   Element Render() override {
     const bool is_focused = Focused();
     const auto focused =
```

Comment:
I think it would be easier to read this way:
```cpp
    const auto focused = (!is_focused && !hovered_) ? select
                         : overwrite()              ? focusCursorBlockBlinking()
                                                    : focusCursorBarBlinking();

```